api/http/handler/registries/proxy_gitlab.go

```diff
@@ -0,0 +1,111 @@
+package registries
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	httperror "github.com/portainer/libhttp/error"
+	"github.com/portainer/libhttp/request"
+	"github.com/portainer/portainer/api"
+)
+
+// request on /api/registries/{id}/proxies/gitlab
+func (handler *Handler) proxyRequestsToGitlabAPIWithRegistry(w http.ResponseWriter, r *http.Request) *httperror.HandlerError {
```

Comment:
I'd rather see each endpoint operation main function relocated to their own file (one entrypoint per operation per file).